cosmos/mocked_dbt_adapters.py

```diff
@@ -0,0 +1,21 @@
+from cosmos.constants import BIGQUERY_PROFILE_TYPE
+
+
+def mock_bigquery_adapter() -> None:
+    from typing import Optional, Tuple
+
+    import agate
+    from dbt.adapters.bigquery.connections import BigQueryAdapterResponse, BigQueryConnectionManager
+    from dbt_common.clients.agate_helper import empty_table
+
+    def execute(  # type: ignore[no-untyped-def]
+        self, sql, auto_begin=False, fetch=None, limit: Optional[int] = None
+    ) -> Tuple[BigQueryAdapterResponse, agate.Table]:
+        return BigQueryAdapterResponse("mock_bigquery_adapter_response"), empty_table()
+
+    BigQueryConnectionManager.execute = execute
```

Comment:
We may have to mock a different method when using `dbt build`. That said, this is outside of the scope of fixing the original but this PR solves, so I've logged a follow up ticket:
https://github.com/astronomer/astronomer-cosmos/issues/1476

Reply:
I agree, thanks for creating the follow-up ticket.